### PR TITLE
[glsl-dependent] Test uninitialized textures are zero when sampled

### PR DESCRIPTION
**Pull request by austinEng**
- need to update @webgpu/types I'll do that on `master` first
 - multisampled tests fail with "Multisampled textures not supported (yet)". That's a Dawn issue. I think we added multisampling support but never removed the validation
 - all other tests pass

### CODE REVIEW

File: src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts [hunk 1]
```diff
@@ -1,73 +1,220 @@
...
+          const actual = await mappedResultBuffer.mapReadAsync();
+          const expected = new readbackTypedArray(new ArrayBuffer(actual.byteLength));
+          expected.fill(expectedShaderValue);
+
+          // TODO: Have a better way to determine approximately equal, maybe in ULPs.
+          let tolerance;
+          if (this.params.format === 'rgb10a2unorm') {
+            tolerance = (i: number) => {
+              // The alpha component is only two bits. Use a generous tolerance.
+              return (i + 1) / 4 === Math.floor((i + 1) / 4) ? 0.18 : 0.01;
```
Comment (kainino0x) [inline src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts:198]: nit:
```suggestion
              return i % 4 === 3 ? 0.18 : 0.01;
```
?

File: src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts [hunk 2]
```diff
@@ -1,73 +1,220 @@
...
+          height *
+          Uint32Array.BYTES_PER_ELEMENT *
+          getTexelDataRepresentation(this.params.format).componentOrder.length;
+        const resultBuffer = this.device.createBuffer({
+          size: byteLength,
+          usage: C.BufferUsage.Storage | C.BufferUsage.CopySrc,
+        });
+
+        const bindGroup = this.device.createBindGroup({
+          // @ts-ignore
```
[the remark this answers is not in the document]
Comment (austinEng) [reply src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts:148]: wouldn't pass the lint check anyway

File: src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts [hunk 3]
```diff
@@ -1,73 +1,220 @@
...
+
+class SampledTextureClearTest extends TextureZeroInitTest {
+  private samplingPipelineCache: Map<string, GPUComputePipeline> = new Map();
+  private getSamplingReadbackPipeline(
+    prefix: string,
+    sampleCount: number,
+    dimension: GPUTextureDimension
+  ): GPUComputePipeline {
+    const MS = sampleCount > 1 ? 'MS' : '';
+    const XD = dimension.toUpperCase();
```
[the remark this answers is not in the document]
Comment (austinEng) [reply src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts]: Done

File: src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (austinEng) [reply src/webgpu/api/operation/resource_init/sampled_texture_clear.spec.ts]: Done

**Pull request merged**